app/models/work-item.ts

```diff
@@ -399,20 +399,27 @@ export async function getWorkItemsByJobIdAndStepIndex(
  * @param notUpdatedForMinutes - jobs with updateAt older than notUpdatedForMinutes ago
  * will be joined with the returned work items
  * @param jobStatus - only jobs with this status will be joined
+ * @param startingId - the work item id to begin the query with, i.e. query work items with id greater than startingId
+ * @param batchSize - the batch size
  * @returns - all work item ids associated with the jobs that
  * met the updatedAt and status constraints
  */
 export async function getWorkItemIdsByJobUpdateAgeAndStatus(
   tx: Transaction,
   notUpdatedForMinutes: number,
   jobStatus: JobStatus[],
+  startingId = 0,
+  batchSize = 2000,
 ): Promise<number[]> {
   const pastDate = subMinutes(new Date(), notUpdatedForMinutes);
   const workItemIds = (await tx(`${WorkItem.table} as w`)
     .innerJoin(Job.table, 'w.jobID', '=', `${Job.table}.jobID`)
-    .select(...tableFields)
+    .select(['w.id'])
```

Comment:
This line and the similar one in `workflow-step.ts` are the most important fixes in this PR. With these two lines of code change, we probably don't need the batching change at all.